pkg/alicloud/secret.go

```diff
@@ -60,11 +71,23 @@ func ReadSecretCredentials(secret *corev1.Secret) (*Credentials, error) {
 }
 
 // ReadCredentialsFromSecretRef reads the credentials from the secret referred by given <secretRef>.
-func ReadCredentialsFromSecretRef(ctx context.Context, client client.Client, secretRef *corev1.SecretReference) (*Credentials, error) {
+func ReadCredentialsFromSecretRef(ctx context.Context, client client.Client, secretRef *corev1.SecretReference, allowDNSKeys bool) (*Credentials, error) {
```

Comment:
I would suggest introduce another function to read DNS credentials only. The name is like `ReadDNSCredentialsFromSecretRef`.The function tries  to read `dnsAccessKeyID / dnsAccessKeySecret` first. If they don't exist, tries `accessKeyID / accessKeySecret` as fallback.

Reply:
I introduced the function `ReadDNSCredentialsFromSecretRef` as you suggested, but the only difference from `ReadCredentialsFromSecretRef` is that is passes `true` instead of false as the last argument to `ReadSecretCredentials`, in this way allowing DNS secret keys in addition to the non-DNS ones. I didn't implement checking for the DNS key first though, since this would mean introducing additional complexity in `ReadSecretCredentials` (or duplicating most of the logic in another function), and the difference would be really minor - both variants of secret keys are already supported, and the order in which they are checked would only matter if we have both e.g. `accessKeyID` and `ACCESS_KEY_ID` in the same secret, which I don't consider to be likely or important enough to make the logic more complex here.

Reply:
Also have in mind that this logic is very similar to what's implemented in all other provider extensions, and it's already merged (or going to be merged soon).

Reply:
Thanks. You are right. I checked the code again. The secret used by DNS is different with that used by infra.

Reply:
Please just keep the same logic as other providers. And also, I think `ReadDNSCredentialsFromSecretRef` is not needed.

Reply:
Since I've already introduced the new function, let's keep it, it's a rather cosmetic change and it's ok this way I think.